onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -615,34 +634,10 @@ Status TensorProtoToTensor(const Env& env, const ORTCHAR_T* model_path,
   void* raw_data = nullptr;
   SafeInt<size_t> raw_data_len = 0;
   AutoDelete deleter_for_file_data;
+  OrtCallback& d = deleter_for_file_data.d;
 
   if (utils::HasExternalData(tensor_proto)) {
-    // Get the external data info
-    std::basic_string<ORTCHAR_T> external_data_file_path;
-    FileOffsetType file_offset;
-    std::basic_string<ORTCHAR_T> tensor_proto_dir;
-    if (model_path != nullptr) {
-      ORT_RETURN_IF_ERROR(GetDirNameFromFilePath(model_path, tensor_proto_dir));
-    }
-    ORT_RETURN_IF_ERROR(GetExternalDataInfo(
```

Comment:
Some test failures are due to this change. Previously the model_path was optional when calling TensorProtoToTensor with external data.

Reply:
Thanks for catching that. I recovered the optional `model_path` and tests are passing now.